modules/frontend/journal/JournalFrontendModule.php

```diff
@@ -87,20 +95,22 @@ private function renderRecentJournalEntry($mJournalId) {
 		return $oTemplate;
 	}
 
-	private function renderRecentJournalCommentTeaser($mJournalId) {
-		$oJournalComment = JournalCommentQuery::create()->excludeUnverified()->joinJournalEntry()->useQuery('JournalEntry')->filterByJournalId($mJournalId)->excludeDraft()->endUse()->mostRecentFirst()->findOne();
+	private function renderRecentJournalCommentTeasers($mJournalId, $iLimit) {
+		$aJournalComments = JournalCommentQuery::create()->excludeUnverified()->joinJournalEntry()->useQuery('JournalEntry')->filterByJournalId($mJournalId)->excludeDraft()->endUse()->mostRecentFirst()->limit($iLimit)->find();
 
-		if($oJournalComment === null) {
+		if($aJournalComments == null) {
 			return null;
 		}
-		$oTemplate = $this->constructTemplate('journal_comment_teaser');
-		$sHref = LinkUtil::link($oJournalComment->getJournalEntry()->getLink($this->oJournalPage)).'#comments';
-		$oTemplate->replaceIdentifier('title', TagWriter::quickTag('a', array('rel' => 'internal', 'href' => $sHref), $oJournalComment->getJournalEntry()->getTitle()));
-		$oTemplate->replaceIdentifier('link_to_detail', $sHref);
-		$oTemplate->replaceIdentifier('created_at', $oJournalComment->getCreatedAt('U'));
-		$oTemplate->replaceIdentifier('name', $oJournalComment->getUsername());
-		$oTemplate->replaceIdentifier('text', $oJournalComment->getText(), null, Template::NO_HTML_ESCAPE);
-		return $oTemplate;
+		foreach ($aJournalComments as $oJournalComment) {
+			$oTemplate = $this->constructTemplate('journal_comment_teaser');
+			$sHref = LinkUtil::link($oJournalComment->getJournalEntry()->getLink($this->oJournalPage)).'#comments';
+			$oTemplate->replaceIdentifier('title', TagWriter::quickTag('a', array('rel' => 'internal', 'href' => $sHref), $oJournalComment->getJournalEntry()->getTitle()));
+			$oTemplate->replaceIdentifier('link_to_detail', $sHref);
+			$oTemplate->replaceIdentifier('created_at', $oJournalComment->getCreatedAt('U'));
+			$oTemplate->replaceIdentifier('name', $oJournalComment->getUsername());
+			$oTemplate->replaceIdentifier('text', $oJournalComment->getText(), null, Template::NO_HTML_ESCAPE);
+			return $oTemplate;
```

Comment:
Are you sure you want to return a template only for the first comment? Why then is the method named `renderRecentJournalCommentTeasers` as if it would potentially render more than one teaser?

Reply:
Oh, thanks, I missed that one. I'll fix it, add a list template to render all comment teasers

Reply:
Fixed c309c01335f81e3950c08afc6542678ed715541d